dbms/src/TiDB/Schema/SchemaSyncService.cpp

```diff
@@ -37,54 +37,92 @@ SchemaSyncService::SchemaSyncService(DB::Context & context_)
     , background_pool(context_.getBackgroundPool())
     , log(&Poco::Logger::get("SchemaSyncService"))
 {
+    // Add task for adding and removing keyspace sync schema tasks.
     handle = background_pool.addTask(
         [&, this] {
-            String stage;
             auto keyspaces = context.getTMTContext().getStorages().getAllKeyspaces();
-            KeyspaceID cur_ks = NullspaceID;
-            bool done_anything = false;
-            try
+            std::unique_lock<std::shared_mutex> lock(ks_map_mutex);
```

Comment:
Make this task function to be two sub-functions `SchemaSyncService::addKeyspaceGCTasks` and `SchemaSyncService::removeStaleKeyspaceTasks` seems better.
It is too long for a lambda function...